### PR TITLE
Adding custom cef binary folder in cmake

### PR DESCRIPTION
**Pull request by alsoussi**

### CODE REVIEW

File: CMakeLists.txt [hunk 1]
```diff
@@ -58,25 +58,39 @@ set(cefArchiveURL https://cef-builds.spotifycdn.com/${cefName}.tar.bz2)
...
-        SHOW_PROGRESS
-    )
-
-    # ... and extract
-    message(STATUS "CEF-CMake: Extracting ${cefArchive}")
-    execute_process(
-        COMMAND ${CMAKE_COMMAND} -E tar xzf ${cefArchive}
-        WORKING_DIRECTORY ${CMAKE_CURRENT_SOURCE_DIR}
-    )
+if(EXISTS ${CEF_BINARY_FOLDER})
```
Comment (sketch34) [inline CMakeLists.txt]: Should this also check if the cmake variable CEF_BINARY_FOLDER is defined and has some value? 

```
if (CEF_BINARY_FOLDER AND EXISTS ${CEF_BINARY_FOLDER})
```

EXISTS says "Behavior is well-defined only for explicit full paths". I think ${CEF_BINARY_FOLDER} might eval to "" if it isn't set.

**Review by sketch34 — CHANGES_REQUESTED** [review]
Looks good! Just one note - see what you think.

File: CMakeLists.txt (hunk 1 above)
Comment (alsoussi) [reply CMakeLists.txt]: Yup nice catch, I also did a similar fix in the download section

File: CMakeLists.txt [hunk 2]
```diff
@@ -58,25 +58,39 @@ set(cefArchiveURL https://cef-builds.spotifycdn.com/${cefName}.tar.bz2)
...
+    set(cefFolder ${CMAKE_CURRENT_SOURCE_DIR})
 endif()
 
+set(cefArchive ${cefFolder}/${cefName}.tar.bz2)
+set(cefArchiveFolder ${cefFolder}/${cefName})
+
+# check if its already there
+if(NOT EXISTS ${cefArchiveFolder})
+    # don't download if the user specified the binary folder
+    if(NOT CEF_BINARY_FOLDER AND NOT EXISTS ${cefArchive})
```
Comment (alsoussi) [inline CMakeLists.txt:74]: @sketch34 did a similar fix here for the var define vs folder existing

**Review by sketch34 — APPROVED** [review]

**Pull request merged**